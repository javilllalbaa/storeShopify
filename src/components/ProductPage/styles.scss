@import "../../index";

.product-container {
  padding: $space-2;
}

@media (max-width: $breakpoint-mobile) {
  .product-container {
    padding: $space-2;
  }
}

@media (min-width: 321px) and (max-width: $breakpoint-mobile-2) {
  .product-container {
    padding: 32px 70px;
    width: 70%;
  }
}

@media (min-width: 376px) and (max-width: $breakpoint-mobile-3) {
  .product-container {
    width: 54%;
    padding: 5% 28%;
  }
}

@media (min-width: $breakpoint-windows) {
  .product-container {
    width: 80%;
    padding: 5% 2%;
  }
}


.product-content {
  display: flex;
  justify-content: space-between;
  @media (max-width: $breakpoint-tablet) {
    display: block !important;
  }
}

.container-img {
  flex: 2 1 auto;
  text-align: end;
  @media (max-width: $breakpoint-tablet) {
    text-align: center !important;
    .product-image {
      margin: 0 !important;
    }
  }
}

.product-image {
  width: 50%;
  margin: 0 10% 0 40%;
}

.product-section {
  flex: 2 1 auto;
}

.product-id {
  font-size: $text-5;
  color: $color-subtitle;
}

.product-attributes {
  font-size: $text-5;
  color: $color-subtitle;
  list-style-type: disc;
  margin: 0;
  padding: 0;
  position: relative;
  left: 18px;
  li {
    margin-bottom: 10px;
  }
}

.product-title {
  font: {
    size: $text-3;
    weight: 500;
  }
  margin: 10px 0;
  margin-top: $space-1;
  color: $color-title;
  width: 400px;
}

.product-amount {
  font: {
    size: $text-1;
    weight: 40%;
  }
  padding-top: $space-2;
}

.product-button {
  width: 100%;
  height: 52px;
  background-color: $button-color;
  border-radius: 6px;
  color: $layout-color;
  font-size: $text-3;
  text-align: center;
  margin-top: $space-2;
  display: flex;
  align-items: center;
}

.product-button-content {
  text-align: center;
  width: 100%;
}

.product-description-title {
  font: {
    size: $text-2;
    weight: 400;
  }
  margin: 0px;
  margin-top: 80px;
}

.product-description {
  margin-top: $space-2;
  width: 60%;
  color: $color-content;
  font-size: $text-4;
}

@media (max-width: $breakpoint-tablet) {
  .product-responsive {
    width: 100%;
  }
}
