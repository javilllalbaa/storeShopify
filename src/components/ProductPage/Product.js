import "./styles.scss";
import img from "../../assets/images/sin-imagen.jpeg";

const Product = ({ product }) => {

  return (
    <div className="product-container">

      {
        product && product.map((att, index) => (
          <div className="product-content" key={index}>
            <div className="container-img">
              <img
                className="product-image product-responsive"
                src={att.image == null ? img : att.image.src}
                alt={att.handle}
              />
            </div>
            <div className="product-section product-responsive">
              <div className="product-id">
                Código de Catálogo: {att.id}
              </div>
              <h1 className="product-title">{att.title}</h1>
              <ul className="product-attributes">
                <li>Precio: {att.variants[0].price}</li>
                <li>Vendedor: {att.vendor}</li>
              </ul>
            </div>
          </div>
        ))
      }

    </div>
  );

};

export default Product