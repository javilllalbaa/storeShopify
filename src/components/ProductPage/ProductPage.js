import PropTypes from "prop-types";
import WithLoadingIndicator from "../WithLoading";
import Product from "./Product"
import "./styles.scss";

const ProductWithLoadingIndicator = WithLoadingIndicator(Product);

const ProductPage = ({ product, isFetchingProduct, selectedProduct }) => {

  return (
    <ProductWithLoadingIndicator
      product={product}
      isFetching={isFetchingProduct || product.length > 0}
      selectedProduct={selectedProduct}
    />
  );
};

ProductPage.propTypes = {
  product: PropTypes.object,
  isFetchingProduct: PropTypes.bool
};

export default ProductPage;
