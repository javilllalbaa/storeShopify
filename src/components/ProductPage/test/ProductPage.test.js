import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

import ProductPage from "../ProductPage";

const testProps = {
    product: [
        {
            "id": 8005666078999,
            "title": "dawn snow",
            "body_html": null,
            "vendor": "test-options-javi",
            "product_type": "",
            "variants": [
                {
                    "id": 44054249144599,
                    "product_id": 8005666078999,
                    "title": "Default Title",
                    "price": "9.41"
                }
            ],
            "image": {
                "id": 39860528251159,
                "product_id": 8005666078999,
                "src": "https://cdn.shopify.com/s/files/1/0678/1684/9687/products/helados.jpg?v=1668597326"
            }
        }
    ],
    isFetchingProduct: false,
    selectedProduct: true
};

jest.mock('./../../../store/products', () => ({
    loadProduct: jest.fn().mockReturnValue(testProps.product)
}));

jest.mock("react-redux", () => {
    return {
        useDispatch: () => jest.fn(),
        useSelector: () => jest.fn()
    };
});

jest.mock("react-router-dom", () => {
    return {
        useLocation: jest.fn().mockReturnValue({ pathname: "/" }),
        useHistory: () => ({
            push: jest.fn(),
        }),
    };
});

describe("ProductPage component", () => {


    describe("WHEN the ProductPage component is render", () => {
        const wrapper = mount(<ProductPage {...testProps} />);

        it("THEN should display ProductPage component", () => {
            expect(wrapper).toMatchSnapshot();
        });

        // const product = wrapper
        //     .find("ComponentWithLoadingIndicator")
        //     .dive()
        //     .find("Product");

        // it("THEN should exist the ProductPage component", () => {
        //     expect(product.exists()).toBeTruthy();
        // });

    });

    describe("WHEN the ProductPage component is render with product", () => {
        const wrapper = shallow(<ProductPage {...testProps} />);

        it("THEN should display ProductPage component", () => {
            expect(wrapper).toMatchSnapshot();
        });

        const product = wrapper
            .find("ComponentWithLoadingIndicator")
            .dive()
            .find("Product");

        it("THEN should exist the ProductPage component", () => {
            expect(product.exists()).toBeTruthy();
        });

    });

    describe("WHEN the ProductPage component is render without product", () => {
        testProps.selectedProduct = false
        const wrapper = shallow(<ProductPage {...testProps} />);

        it("THEN should display ProductPage component without product", () => {
            expect(wrapper).toMatchSnapshot();
        });

        const product = wrapper
            .find("ComponentWithLoadingIndicator")
            .dive()
            .find("HomePage");

        it("THEN should exist the HomePage component", () => {
            expect(product.exists()).toBeTruthy();
        });

    });

});