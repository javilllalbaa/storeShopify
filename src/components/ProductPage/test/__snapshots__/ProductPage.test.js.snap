// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage component WHEN the ProductPage component is render THEN should display ProductPage component 1`] = `ReactWrapper {}`;

exports[`ProductPage component WHEN the ProductPage component is render with product THEN should display ProductPage component 1`] = `ShallowWrapper {}`;

exports[`ProductPage component WHEN the ProductPage component is render without product THEN should display ProductPage component 1`] = `ShallowWrapper {}`;

exports[`ProductPage component WHEN the ProductPage component is render without product THEN should display ProductPage component without product 1`] = `ShallowWrapper {}`;
